modules/tinymce/src/core/test/ts/browser/selection/QuirksSelectionTest.ts

```diff
@@ -0,0 +1,38 @@
+import { UiFinder, Waiter } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { Unicode } from '@ephox/katamari';
+import { PlatformDetection } from '@ephox/sand';
+import { TinyAssertions, TinyDom, TinyHooks } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.selection.QuirksSelectionTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    indent: false,
+    entities: 'raw',
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+  const platform = PlatformDetection.detect();
+
+  it('TINY-11676: pressing enter after clicking on a `p` containg a floating `img` in FireFox should not duplicate the image', async function () {
```

Comment:
We need more tests, when clicking before, on and after a image that is `display: block` for example.

Reply:
I added the after case, but I don't get the second part of your comment:
> on and after a image that is display: block for example.
the image is already `display: block`

Reply:
Sorry I ment `float: left` and `float: right` they also move the caret into the image so two more cases.